lib/myrmidex/generators/approximate.ex

```diff
@@ -151,9 +151,46 @@ defmodule Myrmidex.Generators.Approximate do
   end
 
   defp zip_limits({%mod{}, %mod{}} = limits, opts) do
-    limits
+    case mod do
+      Date -> zip_date_limits(limits)
+      DateTime -> zip_datetime_limits(limits, opts)
+      Time -> zip_time_limits(limits, opts)
+      _ -> zip_generic_limits(limits, opts)
+    end
+  end
+
+  defp zip_date_limits({start_date, end_date}) do
+    [
+      {:year, start_date.year..end_date.year},
+      {:month, start_date.month..end_date.month},
+      {:day, 1..31}
```

Comment:
There's no good solution here as it's a range, but if it's the 15th of January to the 10th of February your range is 1..10 and 15..31, but not 11..14